x-pack/solutions/security/plugins/security_solution/docs/rfcs/detection_response/customized_rule_alert_telemetry.md

```diff
@@ -0,0 +1,299 @@
+
+In order to properly return these fields I believe some refactoring of the `calculateRuleDiff` will need to be done to somehow return these ungrouped fields during the diffing process so that we aren't repeating comparisions needlessly. This work shouldn't need to change the underlying implemetation of the function in any of these endpoints, and with our current test coverage, I think it would be safe enough to refactor this isolated step within the rule diffing process.
+
+### Functional field definition
+A functional field is defined as a field that, when modified, has the ability to change whether or not an alert is generated for the purpose of analyzing how often a rule is generating alerts. There are fields that fully meet these criteria (`query`, `filters`, etc.) but also fields that partially meet this criteria - for instance, the `index` field would obviously change how an alert is written, and changing it would have an impact on what events are queried on, but it's more about changing the rule to suit a user's environment rather than changing the output of the rule itself. For fields like these in this "gray area" between a functional field and a definitively non-functional field (something like `description` which has absolutely no impact on alert generation), we can define as supporting fields: fields that affect rule behavior and operation, but don't change the execution outcome. 
+
+With this definition, we will be conservative in the fields we are defining as "functional" and only the 100%, unarguable fields will be labeled as such. Given we have a list of fields to query on in the telemetry data, alerts can be filtered upon in a more nuanced way on the telemetry side of things if needed.
+
+A full list of fields we diff on with their functional classification is located in the RFC appendix.
+
+### Schema definitions
+
+We will want to map this new field to the existing `rule_source` field mapping. Notably **NOT** as a nested field, as it would constrain us heavily for KQL queries and dashboard tooling later on down the line. 
+
+Putting it all together under one field also makes it very easy to allowlist into the alert telemetry schema they already have without including the rest of the `kibana.alert.rule.parameters` field. Including this whole parameters object could increase the size of the alert document by a lot and wouldn't really add a lot of useful data at scale. Plus, if this data *was* ever needed further down the line, it wouldn't be that difficult to add on the telemetry side of things.
+
+**Mapping example of new rule source object for prebuilt rules**
+
+```js
+rule_source: {
+    type: 'external';
+    is_customized: boolean;
+    customized_fields: Array<{
+        field_name: string;
+    }>;
+    missing_base_version: boolean;
```

Comment:
One thing to think about during the implementation is what should be the default values of the new fields for the rules that already exist.